components/experimental/src/duration/format.rs

```diff
@@ -66,14 +68,76 @@ pub struct FormattedDuration<'l> {
     pub(crate) duration: &'l Duration,
 }
 
+/// Exists to allow creating lists of heterogeneous [`Writeable`]s to pass to [`ListFormatter`].
+/// The (Unit, FixedDecimal) pair is used to crerate [`FormattedUnit`]s.
+type HeterogenousToFormatter = Either<DigitalDurationFormatter, (Unit, FixedDecimal)>;
+
+/// Describes a formatted duration.
+#[derive(Default)]
+struct DigitalDurationFormatter {
```

Comment:
Nit: The name of this would be `DigitalDuration`, since you format it to a `FormattedDigitalDuration`. It is not a `Formatter` because it doesn't own its own data. It owns the thing being formatted.

Reply:
agree